hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/main/java/org/apache/hadoop/mapreduce/jobhistory/JobHistoryEventHandler.java

```diff
@@ -1588,6 +1588,19 @@ public void stop() {
       this.jobIndexInfo =
           new JobIndexInfo(-1, -1, user, jobName, jobId, -1, -1, null,
                            queueName);
+
+      if (getConfig().getBoolean(
+          JHAdminConfig.MR_HISTORY_APPEND_RM_HOST_TO_HISTORY_FILE_NAME_ENABLED,
+          JHAdminConfig.DEFAULT_MR_HISTORY_APPEND_RM_HOST_TO_HISTORY_FILE_NAME_ENABLED)) {
+
+        String hostName = getConfig().get(YarnConfiguration.RM_HOSTNAME, "");
```

Comment:
If we don't config the `yarn.resourcemanager.address` in client, how do we get the right address? And if enabled the federation feature, how can we ensure this?

Reply:
If we don't config it, it will not show up on the UI.

Could you tell me what federation feature you are referring to?

Reply:
I means the YARN Federation feature. As I know in the federation cluster has only one JobHistory server, but maybe has more YARN physical clusters.

Reply:
I see that YARN federation feature can have multiple RMs. 
I created a high availability cluster to check my changes, and noticed that one of the RMs (job-cluster-m-0) was listed as the host. 

Is it safe to assume that YARN federation will behave similarly? Please note that I used Hadoop 3.2.1 to test this.

Reply:
I don't think so. In federation cluster, the job can be route to any physical cluster.